lib/parser/message.h

```diff
@@ -84,8 +88,14 @@ class MessageFormattedText {
   std::string MoveString() { return std::move(string_); }
 
 private:
-  std::string string_;
+  void Format(MessageFixedText text, ...);
+  template<typename A> A Convert(A &&x) { return std::forward<A>(x); }
+  const char *Convert(const std::string &x) { return x.c_str(); }
+  const char *Convert(CharBlock);
+
   bool isFatal_{false};
+  std::string string_;
+  std::vector<std::string> conversions_;  // preserves created strings
```

Comment:
References/pointers to `std::vector` elements are not guaranteed to be valid after new elements are inserted/emplaced. In fact, I caught this because libc++ is resizing the container before inserting the second name in message generated at `lib/semantics/expression.cc:1330`. This resulted in rubbish in the first name such as  `Externally visible object 'ø<80>^C' must ...`
Anyway, using a `std::list` fixes the issue and is guaranteed to preserve the references through its life-time: [cpp ref source](https://en.cppreference.com/w/cpp/container#Iterator_invalidation).